contracts/SL2RD.sol

```diff
@@ -475,4 +481,29 @@ contract SL2RD is
         require(asset.owner() == address(this), "SHARE025");
         asset.transferOwnership(msg.sender);
     }
+
+    /// @notice Sets the ERC20 contract address (e.g., for USDC payments).
+    function setERC20ContractAddress(
+        address contractAddress_
+    ) public override afterInit onlyOwner nonReentrant {
+        require(
+            contractAddress_ != address(0),
+            "Invalid ERC20 contract address"
```

Comment:
@xiangformless Please follow the convention here of using a `SHARE<number>` error code. See [1] & [2] for example.

[1] https://github.com/formless-eng/share-v1-core/blob/9644cd7b695a8fd06169905123e45607f1ebd4e1/contracts/PFA.sol#L185

[2] https://github.com/formless-eng/share-v1-core/blob/9644cd7b695a8fd06169905123e45607f1ebd4e1/README.md?plain=1#L123